lib/core/client.js

```diff
@@ -114,6 +115,21 @@ class NightwatchAPI {
   isOpera() {
     return this.capabilities.browserName === Browser.OPERA;
   }
+
+  isAppiumClient() {
+    if (this.options.selenium && this.options.selenium.use_appium) {
+      return true;
+    }
+
+    // Handle BrowserStack case
+    // (BrowserStack always returns platformName in capabilities)
+    const isMobile = this.__isPlatformName('android') || this.__isPlatformName('ios');
```

Comment:
Use isMobile from utils

Reply:
Cannot use `isMobile` from utils here because that method only considers `desiredCapabilites` whereas over here we need to check is `android` or `ios` is present as `platformName` inside the `capabilities` returned by BrowserStack.

This is necessary because users might not explicitly specify `platformName` in their `desiredCapabilites` (for example, when testing on mobile browser on BrowserStack, `platformName` in desired capabilites is not necessary) but BrowserStack would always send back the `platformName` in `capabilities`.

Reply:
We would also need to tweek `isAndroid()` and `isIOS()` methods available in our API accordingly for it to consider `capabilities` as well, but that is out of scope of this PR and I'll create a separate issue for that.